app/Services/Manager.php

```diff
@@ -254,6 +254,33 @@ public function appendReportback($data, $parameters)
     }
 
     /**
+     * Get the top three reportbacks from a given leaderboard.
+     *
+     * @param array $leaderboard
+     * @return array $topThree
+     */
+    public function getTopThreeReportbacks($leaderboard)
+    {
+        $topThreeUsers = array_slice($leaderboard, 0, 3);
+        $places = ['1st', '2nd', '3rd'];
+
+        foreach ($topThreeUsers as $key => $user) {
+            $reportbackItems = $user->reportback->reportback_items->data;
+            $latestReportbackItem = array_pop($reportbackItems);
+
+            $topThree[] = [
+                'place' => $places[$key],
+                'first_name' => $user->first_name,
+                'quantity' => $user->reportback->quantity,
+                'image_url' => $latestReportbackItem->media->uri,
+                'caption' => $latestReportbackItem->caption,
+            ];
+        }
+
+        return $topThree;
```

Comment:
I think if you don't assign `$topThree = [];` up top, outside the foreach loop, then it very well could fail, if the `$leaderboard` is an empty array and the foreach never runs and never creates the `$topThree` variable? _I think_ 🤔